compiler/optimizer/PreExistence.hpp

```diff
@@ -56,7 +56,7 @@ class TR_PrexArgument
       _profiledClazz(profiledClazz),
       _knownObjectIndex(TR::KnownObjectTable::UNKNOWN),
       _isTypeInfoForInlinedBody(false)
-      { }
+      { TR_ASSERT(_classKind != ClassIsFixed || _class, "Fixed type must have a class"); }
```

Comment:
If this is worth checking - shouldn't it be `TR_ASSERT_FATAL` since the check is cheap?

Reply:
It should be.